apps/sales/views.py

```diff
@@ -154,4 +156,128 @@ def create(self, request, *args, **kwargs):
+        sale_order_items_data = given_data.pop('sale_order_items')
+        order_attachments_data = given_data.pop('order_attachments')
+        order_shipments_data = given_data.pop('order_shipments')
+        
+        # create data in 'saleorder' model
+        serializer = self.get_serializer(data=sale_order_data)
+        if serializer.is_valid(raise_exception=True):
+            self.perform_create(serializer)
+        # if serializer.is_valid():
+        #     serializer.save()
+            print('***SaleOrder-successful***')
+    
+            # create data in 'saleorder_items' model
+            sale_order_id = serializer.data.get('sale_order_id', None) # from Saleorder instance id is fetched
+            add_key_value_to_all_ordereddicts(sale_order_items_data,'sale_order_id',sale_order_id)
+            create_multi_instance(sale_order_items_data,SaleOrderItemsSerializer)
+            print('***SaleOrderItems-successful***')
+
+            # create data in 'order_attachments' model
+            create_multi_instance(order_attachments_data,OrderAttachmentsSerializer)
+            print('***OrderAttachments-successful***')
+
+            # create data in 'order_shipments' model
+            serializer = OrderShipmentsSerializer(data=order_shipments_data)
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()        
+            print('***\tAll Instances are Created***')
+
+            return Response({sale_order_id})
+
+            # return Response({'***data ceated successfully***'})
+
+    def retrieve(self, request, *args, **kwargs):
+        pk = kwargs.get('pk')  # Access the pk from kwargs
+        instance = self.get_object()
+        serializer = self.get_serializer(instance)
+
+        # Query SaleOrderItems model using the pk
+        items_related_data = SaleOrderItems.objects.filter(sale_order_id=pk)  # 'sale_order_id' is the FK field
+        items_related_serializer = SaleOrderItemsSerializer(items_related_data, many=True)
+
+        # get order_id value from SaleOrder Instance 
+        order_id = serializer.data.get('order_no')
+
+        # Query OrderAttachments model using the order_id
+        attachments_related_data = OrderAttachments.objects.filter(order_id=str(order_id))
+        attachments_related_serializer = OrderAttachmentsSerializer(attachments_related_data, many=True)
+
+        # Query OrderShipments model using the order_id
+        shipments_related_data = OrderShipments.objects.filter(order_id=str(order_id))
+        shipments_related_serializer = OrderShipmentsSerializer(shipments_related_data, many=True)
+
+        # Customizing the response data
+        custom_data = {
+            "sale_order": serializer.data,
+            "sale_order_items": items_related_serializer.data,
+            "order_attachments":attachments_related_serializer.data,
+            "order_shipments": shipments_related_serializer.data,
+        }
+        return Response(custom_data)
```

Comment:
The custom_data JSON should be similar to input JSON that we get for POST call

Reply:
Updated Thulasi